protoc-gen-gogo/generator/generator.go

```diff
@@ -3004,6 +3005,7 @@ func (g *Generator) generateOneofDecls(mc *msgCtx, topLevelFields []topLevelFiel
 		if gogoproto.IsProtoSizer(g.file.FileDescriptorProto, mc.message.DescriptorProto) {
 			g.P(`ProtoSize() int`)
 		}
+		g.P("Clone()", of.goType)
```

Comment:
Use `dname` here?

Reply:
Also, suggest to track all changes - ideally we would make this depend on a flag before we attempt to upstream (which we should)